scipy/stats/_hypotests.py

```diff
@@ -1262,31 +1262,42 @@ def _get_binomial_log_p_value_with_nuisance_param(
     return -log_pvalue
 
 
-def _pval_cvm_2samp_exact(s, nx, ny):
+def _pval_cvm_2samp_exact(s, m, n):
     """
     Compute the exact p-value of the Cramer-von Mises two-sample test
-    for a given value s (float) of the test statistic by enumerating
-    all possible combinations. nx and ny are the sizes of the samples.
-    """
-    rangex = np.arange(nx)
-    rangey = np.arange(ny)
-
-    us = []
+    for a given value s of the test statistic.
+    m and n are the sizes of the samples.
 
-    # x and y are all possible partitions of ranks from 0 to nx + ny - 1
-    # into two sets of length nx and ny
-    # Here, ranks are from 0 to nx + ny - 1 instead of 1 to nx + ny, but
-    # this does not change the value of the statistic.
-    for x, y in _all_partitions(nx, ny):
-        # compute the statistic
-        u = nx * np.sum((x - rangex)**2)
-        u += ny * np.sum((y - rangey)**2)
-        us.append(u)
-
-    # compute the values of u and the frequencies
-    u, cnt = np.unique(us, return_counts=True)
-    return np.sum(cnt[u >= s]) / np.sum(cnt)
+    [1] Y. Xiao, A. Gordon, and A. Yakovlev, “A C++ Program for the Cramér-Von Mises Two-Sample Test”, J. Stat. Soft., vol. 17, no. 8, pp. 1–15, Dec. 2006.
+    [2] T. W. Anderson "On the Distribution of the Two-Sample Cramer-von Mises Criterion," The Annals of Mathematical Statistics, Ann. Math. Statist. 33(3), 1148-1159, (September, 1962)
+    """
 
+    # [1, p. 3]
+    l = math.lcm(m, n)
```

Comment:
Looks like the test suite is complaining that `math` hasn't been imported yet.

I do have a more useful comment though--`math.lcm` was added in Python `3.9`, so you may need to work around that for a little while, until we drop support for `3.8` at least, assuming this remains in the code.

Scanning around the internet for ideas, it sounds like `numpy.lcm` has been around for quite some time, so that might be an option.